plugins/GSdx/GSTextureCache.cpp

```diff
@@ -867,7 +849,7 @@ void GSTextureCache::InvalidateVideoMem(GSOffset* off, const GSVector4i& rect, b
 				}
 				else
 				{
-					m_dst[type].erase(j);
+					i = list.erase(i);
```

Comment:
honestly I found your pattern very error-prone. For short loop it is fine but here code is quite complete and you need to read it carefully (if future branch are created/delete) to properly update the iterator. Speed wise the saving is 0, so for me it isn't a good idea here